### PR TITLE
craete the backend for google calendar

### PR DESCRIPTION
**Pull request by Bakeza**
Relates #103

### CODE REVIEW

**Review by shamskhodary — APPROVED** [review]

**Review by SaifHayek — APPROVED** [review]

**Review by SaifHayek — DISMISSED** [review]

File: client/src/App.tsx [hunk 1]
```diff
@@ -61,6 +62,10 @@ const App: React.FC = () => {
       path: 'calendar',
       element: !auth.user ? <Navigate to="/" replace /> : <Calender />,
     },
+    // {
+    //   path: 'GoogleCalendar',
+    //   element: <GoogleCalendar />,
+    // },
```
Comment (aaamra) [inline client/src/App.tsx:68]: remove the comments

File: client/src/components/AddEvent/index.tsx [hunk 2]
```diff
@@ -271,6 +272,20 @@ const AddEvent: FC<addEventProps> = ({ setIsAdded }) => {
...
+                'Check out the maybe easiest way to include add to calendar buttons to your website at:<br>→ [url]https://github.com/add2cal/add-to-calendar-button[/url]',
+                startDate: 'today+2',
+                endDate: 'today+4',
+                location: 'https://github.com/add2cal/add-to-calendar-button',
+                options: [
+                  'Google',
+                ],
+                timeZone: 'Europe/Berlin',
+              }}
+            </div> */}
```
Comment (aaamra) [inline client/src/components/AddEvent/index.tsx:288]: remove the comments

File: client/src/components/GoogleCalendar/index.tsx [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+import {
+  Dialog, DialogTitle, DialogContentText, DialogContent, DialogActions,
+} from '@mui/material'
+import { toast } from 'react-toastify'
+
+const GCalendar: FC<{getEventDataForCalendar:Function, disabled: boolean}> = (
+  { getEventDataForCalendar, disabled },
+) => {
+  const [open, setOpen] = React.useState(false)
+  // const [openOnce, setOpenOnce] = React.useState(true)
```
Comment (aaamra) [inline client/src/components/GoogleCalendar/index.tsx:13]: remove this comment

File: src/controllers/GoogleCalendar.ts [hunk 4]
```diff
@@ -0,0 +1,90 @@
+import { Response } from 'express'
+import { google } from 'googleapis'
+import { JoinedPeople, User } from '../db'
+// import validateParams from '../validation/paramsId'
+import { IUserRequest } from '../interfaces/IUserRequest'
+import config from '../config/environment'
+
+const Credentials = async (req:IUserRequest, res:Response):Promise<void> => {
```
Comment (aaamra) [inline src/controllers/GoogleCalendar.ts:8]: dont forget the validation

File: src/helpers/expressWrapper.ts [hunk 5]
```diff
@@ -7,6 +7,7 @@ const expressWrapper = (controller: Function) => {
     try {
       await controller(req, res, next)
     } catch (err: any) {
+      console.log(err)
```
Comment (aaamra) [inline src/helpers/expressWrapper.ts:10]: remove this

File: src/routes/events/index.ts [hunk 6]
```diff
@@ -19,6 +20,8 @@ router.get('/events', expressWrapper(EventsController.index))
 router.get('/events/joined', expressWrapper(isAuth), expressWrapper(CalendarJoinedController.index))
 
 router.get('/events/interested', expressWrapper(isAuth), expressWrapper(CalendarInterestedController.index))
+router.post('/events/googleCalendar', expressWrapper(isAuth), expressWrapper(GoogleCalendar))
+// router.get('/events/googleCalendar/:id', expressWrapper(GetEventId))
```
Comment (aaamra) [inline src/routes/events/index.ts:24]: remove this comment

**Review by aaamra — CHANGES_REQUESTED** [review]

**Review by shamskhodary — DISMISSED** [review]

**Review by SaifHayek — APPROVED** [review]

**Review by MostafaBalousha123 — DISMISSED** [review]

**Review by aaamra — APPROVED** [review]

**Pull request merged**